stellargraph/data/unsupervisedSampler.py

```diff
@@ -0,0 +1,229 @@
+
+from stellargraph.core.utils import is_real_iterable
+from stellargraph.core.graph import StellarGraphBase
+from stellargraph.data.explorer import UniformRandomWalk
+
+
+class UnsupervisedSampler:
+    """
+        This class does unsupervised samlping in a graph. 
+        As the minimum, tt takes a Stellargraph graph and and walker instance. The walker is the the graph sampling strategy, such as uniform random walk, biased random walk etc.
+        At the moment only uniform random walks are performed but this can be easily modified to accomodate other types of walks.
+        Given a graph and walker, the generator method of this Unsupervisedsampler generates a batch of positive and negative samples. The generator method requires a batch size.
+        The positive samples are all the (target, context) pairs from the walks generated by walker. Negative samples are contexts generated for each target based on a sampling strategy.
+    """
+
+    def __init__(
+        self, G, walker=None, nodes=None, length=2, number_of_walks=1, seed=None
+    ):
+
+        """
+           Initializing all the paramters for the class instance. A Stellargraph is a must. 
+           The root nodes and walker are required. However, if not provided they default to all nodes in the graph and uniform random walker respectively.
+           The walks should have length at least 2. Need 2 nodes for a valid taget,contect pair.
+           At least one walk should be performed from each root node. Actually number of walks can be completely ignored here. Since in the end we just need the batch size to generate samples. 
+           So if for example, is large, we will do as many walks as needed to generate enough samples. The reason this parameter is still needed is to estimate how many batches in each epoch
+           are generated. 
+           
+           Args:
+               G <StellarGraph>: A stellargraph is a must.
+               walker <UniformRandomWalker>: The only walking method supported for sampling is Uniform Random Walk.
+               nodes <iterable of nodes from the graph>: if none provided, defaults to all nodes in the graph as root nodes for walks.
+               length <int>: An integer value for the length of the walk. Minimum length is 2 since the samples are a pair of target,context so at least one hop is needed beyond the root node.
+               number_of_walks <int>: Number of walks from each root node. Note, at the moment this parameter is only used to calculate the number of batches in each epoch,
+                                      as we do not have a pregenerated sample list, we need to estimate given a batch_size, how many batches to generate in each epoch.  
+               
+        
+        """
+        # Initialize the random state
+        self._random_state = random.Random(seed)
+
+        # Initialize a numpy random state (for numpy random methods)
+        self._np_random_state = np.random.RandomState(seed=seed)
```

Comment:
You never use these random seeds. Remove them if unneeded.

Reply:
Removed them.